jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2604,4 +2603,21 @@ else if (oldQuery == null)
     {
         throw new ServletException("HttpServletRequest.upgrade() not supported in Jetty");
     }
+
+    private String formatAddrOrHost(String addr)
+    {
+        if (addr == null)
+            return addr;
+        if (getContext() == null || getContext().getContextHandler() == null)
+            return HostPort.normalizeHost(addr);
+        switch (getContext().getContextHandler().getIpv6Format())
+        {
+            case BRACKETED:
+                return HostPort.normalizeHost(addr);
+            case UNBRACKETED:
+                return HostPort.denormalizeHost(addr);
+            default:
```

Comment:
I'd rather have the explicit case, and a `default` case that throws, in case in future we add other formats.